### PR TITLE
Add support for displaying test suite results in perfdash

### PR DESCRIPTION
**Pull request by pierewoj**
This commit enables perfdash to view metrics recorded by clusterloader
test suite runs.

Before this change, it was not possible to do so, because test suite run
puts measurement results in separate files, for example:
-  APIResponsiveness_storage_configmap-vol-per-node_2019-10-01T06:08:05Z.json
-  APIResponsiveness_storage_downwardapi-vol-per-node_2019-10-01T06:10:00Z.json
In such case, perfdash read metrics data from the first file that
matched the prefix APIResponsiveness_storage (prefix is specified in
config.go file), meaning that other results were ignored.

To solve this issue, support for handling multiple files that match
given prefix is added. For each of the matched files, perfdash extracts
the "suiteId" from the file name and appends it to the result category.
This means that results from different test cases in a given suite will
land in different categories in the web UI, for example:
- APIServer_configmap-vol-per-node
- APIServer_downwardapi-vol-per-node

Alternatively, same effect could be achieved by generating separate test
descriptions in the perfdash config. It was not chosen, because it would
mean that perfdash config needs to be changed every time some test case
is added or removed.

This commit also includes other minor, but related changes like:
- increasing with of category field in the web UI
- removing 'pod-with-ephemeral-volume' configs

Testing:
- run perfdash locally, verify that data for both storage-suite and
separate storage jobs is displayed as expected

### CODE REVIEW

File: perfdash/config.go [hunk 1]
```diff
@@ -282,24 +282,14 @@ var (
 		"APIServer": {
 			"Responsiveness": []TestDescription{
 				{
-					Name:             "pod-with-ephemeral-volume-startup-latency",
-					OutputFilePrefix: "APIResponsiveness",
-					Parser:           parsePerfData,
-				},
-				{
-					Name:             "storage-",
+					Name:             "storage",
```
Comment (mborsz) [inline perfdash/config.go]: Do the "old storage" test still work after this change?

Comment (pierewoj) [reply perfdash/config.go]: It seems to work properly:
* I've started the perfdash locally and it has both the "old storage" metrics and "storage suite" metrics
* Checked files from old test run (https://gcsweb.k8s.io/gcs/kubernetes-jenkins/logs/ci-kubernetes-storage-scalability-max-downwardapi-vol-per-node/1178934805790724098/artifacts/) and each it looks like using "storage-" or "storage" prefixes should have same effect

File: perfdash/google-gcs-downloader.go [hunk 2]
```diff
@@ -120,33 +114,57 @@ func (g *GoogleGCSDownloader) getJobData(wg *sync.WaitGroup, result JobToCategor
...
 					if err != nil || len(artifacts) == 0 {
-						fmt.Printf("Error while looking for %s* in %s build %v: %v\n", fileStem, job, buildNumber, err)
+						fmt.Printf("Error while looking for %s* in %s build %v: %v\n", searchPrefix, job, buildNumber, err)
 						continue
 					}
-					metricsFilename := artifacts[0][strings.LastIndex(artifacts[0], "/")+1:]
-					if len(artifacts) > 1 {
-						fmt.Printf("WARNING: found multiple %s files with data, reading only one: %s\n", fileStem, metricsFilename)
+					for _, artifact := range artifacts {
+						metricsFileName := artifact[strings.LastIndex(artifact, "/")+1:]
```
Comment (mborsz) [inline perfdash/google-gcs-downloader.go]: Please use https://golang.org/pkg/path/filepath/#Base -- it will be easier to read IMHO.

File: perfdash/google-gcs-downloader.go [hunk 3]
```diff
@@ -120,33 +114,57 @@ func (g *GoogleGCSDownloader) getJobData(wg *sync.WaitGroup, result JobToCategor
...
 	}
 }
 
+func getResultCategory(metricsFileName string, filePrefix string, category string, artifacts []string) string {
+	if len(artifacts) <= 1 {
+		return category
+	}
+	// if there are more artifacts, assume that this is a test suite run
+	trimmed := strings.TrimPrefix(metricsFileName, filePrefix+"_")
+	suiteId := strings.Split(trimmed, "_")[0]
```
Comment (mborsz) [inline perfdash/google-gcs-downloader.go]: What if len(strings.Split(trimmed, "_")) == 0?

Comment (pierewoj) [reply perfdash/google-gcs-downloader.go]: from split documentation:
```
// If s does not contain sep and sep is not empty, Split returns a
// slice of length 1 whose only element is s.
(...)
func Split(s, sep string) []string { return genSplit(s, sep, 0, -1) }
```

File: perfdash/google-gcs-downloader.go [hunk 4]
```diff
@@ -120,33 +114,57 @@ func (g *GoogleGCSDownloader) getJobData(wg *sync.WaitGroup, result JobToCategor
 		for categoryLabel, categoryMap := range tests.Descriptions {
 			for testLabel, testDescriptions := range categoryMap {
 				for _, testDescription := range testDescriptions {
-					fileStem := fmt.Sprintf("artifacts/%v_%v", testDescription.OutputFilePrefix, testDescription.Name)
-					artifacts, err := g.GoogleGCSBucketUtils.listFilesInBuild(job, buildNumber, fileStem)
+					filePrefix := fmt.Sprintf("%v_%v", testDescription.OutputFilePrefix, testDescription.Name)
+					searchPrefix := fmt.Sprintf("artifacts/%v", filePrefix)
```
Comment (mborsz) [inline perfdash/google-gcs-downloader.go]: Nit: Why this change? I think I preferred the previous version.

File: perfdash/google-gcs-downloader.go (hunk 2 above)
Comment (pierewoj) [reply perfdash/google-gcs-downloader.go]: good tip, fixed

File: perfdash/google-gcs-downloader.go (hunk 4 above)
Comment (pierewoj) [reply perfdash/google-gcs-downloader.go]: mostly because `filePrefix` is used later on when constructing the `resultCategory`, I wanted to avoid building the prefix twice so that we cna change it in one place (hopefully) if the filename conventions change

File: perfdash/google-gcs-downloader.go [hunk 5]
```diff
@@ -120,33 +115,57 @@ func (g *GoogleGCSDownloader) getJobData(wg *sync.WaitGroup, result JobToCategor
...
+					for _, artifact := range artifacts {
+						metricsFileName := filepath.Base(artifact)
+						resultCategory := getResultCategory(metricsFileName, filePrefix, categoryLabel, artifacts)
+						testDataResponse, err := g.GoogleGCSBucketUtils.getFileFromJenkinsGoogleBucket(job, buildNumber,
+							fmt.Sprintf("artifacts/%v", metricsFileName))
+						if err != nil {
+							fmt.Fprintf(os.Stderr, "Error when reading response Body: %v\n", err)
+							continue
+						}
+						buildData := getBuildData(result, tests.Prefix, resultCategory, testLabel, job, resultLock)
```
Comment (mborsz) [inline perfdash/google-gcs-downloader.go]: Don't you need to hold resultLock here?

File: perfdash/google-gcs-downloader.go (hunk 3 above)
Comment (mborsz) [reply perfdash/google-gcs-downloader.go]: Good point.

File: perfdash/google-gcs-downloader.go (hunk 4 above)
Comment (mborsz) [reply perfdash/google-gcs-downloader.go]: OK, makes sense.

File: perfdash/google-gcs-downloader.go (hunk 5 above)
Comment (pierewoj) [reply perfdash/google-gcs-downloader.go]: We actually hold the result lock when accessing the map, see the `getBuildData` impl

Comment (mborsz) [reply perfdash/google-gcs-downloader.go]: You are right. It's OK.

File: perfdash/google-gcs-downloader.go (hunk 3 above)
Comment (krzysied) [reply perfdash/google-gcs-downloader.go]: Can you give example of a file name and how will it behave when there is no suiteId?
file name  = "metric_test_timestamp" => suiteId = timestamp?

File: perfdash/google-gcs-downloader.go [hunk 6]
```diff
@@ -120,33 +115,57 @@ func (g *GoogleGCSDownloader) getJobData(wg *sync.WaitGroup, result JobToCategor
...
 			}
 		}
 	}
 }
 
+func getResultCategory(metricsFileName string, filePrefix string, category string, artifacts []string) string {
+	if len(artifacts) <= 1 {
+		return category
+	}
+	// if there are more artifacts, assume that this is a test suite run
```
Comment (krzysied) [inline perfdash/google-gcs-downloader.go]: nit: Capital letter and period.

File: perfdash/google-gcs-downloader.go (hunk 3 above)
Comment (pierewoj) [reply perfdash/google-gcs-downloader.go]: Yes, that's correct. But this would mean that we'll have two files with same "metric_test" prefix and a different "timestamp" for the same run.

We could add some validation around it. For example we could expect that 
`len(strings.Split(trimmed, "_")) ` is greater than 2. However, I suggest to keep it simple for now. What do you think?

File: perfdash/google-gcs-downloader.go (hunk 6 above)
Comment (pierewoj) [reply perfdash/google-gcs-downloader.go]: Fixed

File: perfdash/google-gcs-downloader.go (hunk 3 above)
Comment (krzysied) [reply perfdash/google-gcs-downloader.go]: > Yes, that's correct. But this would mean that we'll have two files with same "metric_test" prefix and a different "timestamp" for the same run.

What if there is only one file with this name format? Aren't all of our tests creating files like this unless we migrate them to test suite?
Example `ResourceUsageSummary_density_2019-10-02T10:08:21Z.json`

Comment (krzysied) [reply perfdash/google-gcs-downloader.go]: For some reason I missed 
```
if len(artifacts) <= 1 {
	return category
}
```
It should behave correctly for old summaries.
